javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java

```diff
@@ -36,4 +39,12 @@ default VariableDeclarator getVariable(int i) {
     }
 
     N setVariables(NodeList<VariableDeclarator> variables);
+
+    public default Type getCommonType() {
+        return getVariables().stream().map(v -> v.getType())
+                .sorted((Comparator<Type<?>>) (a, b) -> a.getArrayLevel() - b.getArrayLevel())
+                .findFirst()
```

Comment:
Damned annotations :D
Well, this seems an interesting problem that the guy working on #631 will get fixed for me :D